examples/data_preprocessing/mimic_dataset/prepare_mimic_cxr.ipynb

```diff
@@ -0,0 +1,1520 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "train-val-test split proportions: split\n",
+      "test        0.013680\n",
+      "train       0.978388\n",
+      "validate    0.007931\n",
+      "dtype: float64\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"train-val-test split proportions:\" ,record_report_split_label_list.groupby('split').size()/len(record_report_split_label_list))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There are 14 classes, where about 42% of the studies are assigned to the class 'No finding'. We should keep in mind that we are dealing with an unbalanced dataset.\n",
+    "\n",
+    "Now the labels get replaced by their id. This gives us the final \"input_list\". Since we will need this file multiple times and the runtime is relatively long, it is cached."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_list_full = record_report_split_label_list"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['input_list.lib']"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#save the whole file\n",
+    "dump(input_list_full, \"input_list.lib\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will continue working with the input list of the first n rows where the rows including missing values are dropped. Instead of n=1000 we now have n=932"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "932\n"
+     ]
+    }
+   ],
+   "source": [
+    "#open only first n rows\n",
+    "input_list_pd = load(\"input_list.lib\").iloc[:n,:]\n",
+    "#drop nas\n",
+    "input_list = input_list_pd.dropna().to_numpy()\n",
+    "#save new n\n",
+    "n = len(input_list)\n",
+    "print(n)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating rules from Chexpert-labler"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We are using the [\"mentions\" from Chexpert-labler](https://github.com/stanfordmlgroup/chexpert-labeler/tree/master/phrases/mention) for building our rules. Therefore, we need to download the .txt files corresponding to each class from github. The file names are the labels we defined above, where all letters are lowercase and instead of whitespaces there are underscores. So first we do some transformations to get a list of our classes. "
```

Comment:
```suggestion
    "We are using the [\"mentions\" from Chexpert-labler](https://github.com/stanfordmlgroup/chexpert-labeler/tree/master/phrases/mention) for building our rules. Therefore, we need to download the TXT files corresponding to each class from GitHub. The file names are the labels we defined above, where all letters are lowercase and instead of whitespaces, there are underscores. So first we do some transformations to get a list of our classes. "
```

+ mention somehow that our rules are keywords
+ mention that you rules are for annotation of the reports in order to *later* use them to calculate labels, which, in turn, can be later used to annotate images. Now this whole process and the reasons why you need rules is not clear.

Reply:
Is this now clear with the description and graph at the beginning?